### PR TITLE
add condition bind district by adherent country

### PR DESCRIPTION
**Pull request by Raulnet**
add condition bind district by adherent country

### CODE REVIEW

File: src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php [hunk 1]
```diff
@@ -28,17 +28,19 @@ public function updateReferentTagWithDistrict(AdherentEvent $event): void
         $adherent = $event->getAdherent();
 
         if ($adherent->isGeocoded()) {
-            $referentTags = $this->districtRepository->findDistrictReferentTagByCoordinates(
+            $districts = $this->districtRepository->findDistrictReferentTagByCoordinates(
```
Comment (Remg) [inline src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php]: should this method be renamed with something like `findDistrictsByCoordinates($lat, $long)` then ?

Comment (Raulnet) [reply src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php]:
```suggestion
            $districts = $this->districtRepository->findDistrictReferentTagByCoordinates(
```

File: src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php [hunk 2]
```diff
@@ -28,17 +28,19 @@ public function updateReferentTagWithDistrict(AdherentEvent $event): void
...
                 $adherent->getLatitude(),
                 $adherent->getLongitude()
             );
 
-            if (!empty($referentTags)) {
-                foreach ($referentTags as $referentTag) {
-                    if ($adherent->getReferentTags()->contains($referentTag)) {
+            if (!empty($districts)) {
+                foreach ($districts as $district) {
+                    if (!\in_array($adherent->getCountry(), $district->getCountries(), false)) {
```
Comment (ottaviano) [inline src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php]: `false` it's a default value of in_array

File: src/Repository/DistrictRepository.php [hunk 3]
```diff
@@ -17,22 +15,20 @@ public function __construct(RegistryInterface $registry)
...
-    public function findDistrictReferentTagByCoordinates($latitude, $longitude): array
+    public function findDistrictsByCoordinates($latitude, $longitude): array
     {
-        return $this->getEntityManager()->createQueryBuilder()
-            ->from(ReferentTag::class, 'tag')
-            ->select('tag')
-            ->innerJoin(District::class, 'district', Join::WITH, 'district.referentTag = tag')
-            ->innerJoin('district.geoData', 'geoData')
+        return $this->createQueryBuilder('district')
+            ->join('district.referentTag', 'referentTag')
```
Comment (ottaviano) [inline src/Repository/DistrictRepository.php]: can you add `addSelect('referentTag')` for avoiding a additionnal query

File: src/Repository/DistrictRepository.php [hunk 4]
```diff
@@ -17,22 +15,20 @@ public function __construct(RegistryInterface $registry)
...
+            ->join('district.geoData', 'geoData')
             ->where("ST_Within(ST_GeomFromText(CONCAT('POINT(',:longitude,' ',:latitude,')')), geoData.geoShape) = 1")
             ->setParameter('latitude', $latitude)
             ->setParameter('longitude', $longitude)
             ->getQuery()
             ->getResult()
-        ;
+            ;
```
Comment (ottaviano) [inline src/Repository/DistrictRepository.php]: CS: not need

Comment (Raulnet) [reply src/Repository/DistrictRepository.php]: resolved

File: src/Repository/DistrictRepository.php (hunk 3 above)
Comment (Raulnet) [reply src/Repository/DistrictRepository.php]: resolved

File: src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php (hunk 2 above)
Comment (Raulnet) [reply src/Deputy/Subscriber/BindAdherentDistrictSubscriber.php]: resolved

**Pull request merged**